src/devices/video/vrender0.cpp

```diff
@@ -163,119 +224,88 @@ void vr0video_device::device_start()
 //  device_reset - device-specific reset
 //-------------------------------------------------
 
 void vr0video_device::device_reset()
 {
-	memset(m_InternalPalette, 0, sizeof(m_InternalPalette));
-	m_LastPalUpdate = 0xffffffff;
+	std::fill(std::begin(m_internal_palette), std::end(m_internal_palette), 0);
+	m_last_pal_update = 0xffffffff;
 
-	m_DisplayDest = m_DrawDest = m_frameram;
+	m_display_dest = m_draw_dest = 0;
 	// 1100 objects per second at ~80 MHz
-	m_pipeline_timer->adjust(attotime::from_hz(this->clock() /1100), 0, attotime::from_hz(this->clock() / 1100));
+	m_pipeline_timer->adjust(attotime::from_hz(this->clock() / 1100), 0, attotime::from_hz(this->clock() / 1100));
+}
+
+//-------------------------------------------------
+//  memory_space_config - return a description of
+//  any address spaces owned by this device
+//-------------------------------------------------
+
+device_memory_interface::space_config_vector vr0video_device::memory_space_config() const
+{
+	return space_config_vector {
+		std::make_pair(AS_TEXTURE, &m_texture_config),
+		std::make_pair(AS_FRAME, &m_frame_config)
+	};
 }
 
 /*****************************************************************************
  IMPLEMENTATION
  *****************************************************************************/
 
-struct QuadInfo
-{
-	uint16_t *Dest;
-	uint32_t Pitch;   //in UINT16s
-	uint32_t w,h;
-	uint32_t Tx;
-	uint32_t Ty;
-	uint32_t Txdx;
-	uint32_t Tydx;
-	uint32_t Txdy;
-	uint32_t Tydy;
-	uint16_t TWidth;
-	uint16_t THeight;
-	union _u
-	{
-		uint8_t *Imageb;
-		uint16_t *Imagew;
-	} u;
-	uint16_t *Tile;
-	uint16_t *Pal;
-	uint32_t TransColor;
-	uint32_t Shade;
-	uint8_t Clamp;
-	uint8_t Trans;
-	uint8_t SrcAlpha;
-	uint32_t SrcColor;
-	uint8_t DstAlpha;
-	uint32_t DstColor;
-};
-
-/*
-Pick a rare enough color to disable transparency (that way I save a cmp per loop to check
-if I must draw transparent or not. The palette build will take this color in account so
-no color in the palette will have this value
-*/
-#define NOTRANSCOLOR    0xecda
-
-#define RGB32(r,g,b) ((r << 16) | (g << 8) | (b << 0))
-#define RGB16(r,g,b) ((r & 0xf8) << 8) | ((g & 0xfc) << 3) | ((b & 0xf8) >> 3)
-
-static inline uint16_t RGB32TO16(uint32_t rgb)
+constexpr u32 RGB32(u8 r, u8 g, u8 b) { return (r << 16) | (g << 8) | (b << 0); }
+constexpr u16 RGB16(u8 r, u8 g, u8 b) { return ((r & 0xf8) << 8) | ((g & 0xfc) << 3) | ((b & 0xf8) >> 3); }
+
+constexpr u16 RGB32TO16(u32 rgb)
 {
 	return (((rgb >> (16 + 3)) & 0x1f) << 11) | (((rgb >> (8 + 2)) & 0x3f) << 5) | (((rgb >> (3)) & 0x1f) << 0);
 }
 
-#define EXTRACTR8(Src)  (((Src >> 11) << 3) & 0xff)
-#define EXTRACTG8(Src)  (((Src >>  5) << 2) & 0xff)
-#define EXTRACTB8(Src)  (((Src >>  0) << 3) & 0xff)
+constexpr u8 EXTRACTR8(u16 src) { return ((src >> 11) << 3) & 0xff; }
+constexpr u8 EXTRACTG8(u16 src) { return ((src >>  5) << 2) & 0xff; }
+constexpr u8 EXTRACTB8(u16 src) { return ((src >>  0) << 3) & 0xff; }
```

Comment:
These end up only setting five or six out of eight bits – should they use the `pal5bit` and `pal6bit` helpers that splat right instead?